src/state/slices/foxEthSlice/contractManager.ts

```diff
@@ -0,0 +1,23 @@
+import type { ContractInterface } from '@ethersproject/contracts'
+import { Contract } from '@ethersproject/contracts'
+
+import { ethersProvider } from './utils'
+
+type DefinedContract = {
+  contract: Contract
+  address: string
+  abi: ContractInterface
+}
+
+const definedContracts: DefinedContract[] = []
+
+export const getOrCreateContract = (address: string, abi: ContractInterface): Contract => {
+  const definedContract = definedContracts.find(
+    contract => contract.address === address && contract.abi === abi,
```

Comment:
`contract.abi === abi` will compare the abi from `definedContracts` to the one you're passing to it and is not reliable, since every import of the JSON will give you a new object, thus a new reference and comparing objects is done by value.

You can either compared the stringified value or use deep equal comparison to achieve this.

Alternatively, you might want to ditch the check altogether: the ABI for a contract at a given address is guaranteed to never change, so there should be no need to compare ABIs in the first place.